#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.014274313778353087 0.9961331839332513 -0.08668866034932393 2.886345788013788
  position -0.07353489090429978 0.09588180379542949 -2.823066800413585
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 50 50
}
Robot {
  children [
    DEF Body Transform {
      translation 0 0.6 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.5 0.5
          }
        }
      ]
    }
    DEF FrontLeft_upper HingeJoint {
      jointParameters HingeJointParameters {
        position -0.7
        axis 0 0 1
        anchor -0.16 0.38 0.3
      }
      device [
        RotationalMotor {
          name "FrontLeft_Shoulder"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.20388923272359366 0.3122999707833206 0.29999973462432733
        rotation 5.374765207010968e-06 1.9618496973603487e-06 -0.9999999999836316 0.7000035302572778
        children [
          DEF FrontLeft_lower HingeJoint {
            jointParameters HingeJointParameters {
              position 1.1
              axis 0 0 1
              anchor 0 -0.05 -0.3
            }
            device [
              RotationalMotor {
                name "FrontLeft_Elbow"
                maxTorque 100
              }
            ]
            endPoint Solid {
              translation 0.09803280960675789 -0.09989557335681357 0.09999999999999998
              rotation 0 0 1 1.1000000000000003
              children [
                DEF FL_lowerLeg Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.25 0.1
                  }
                }
              ]
              name "solid(1)"
              boundingObject USE FL_lowerLeg
              physics DEF lowerLeg_physics Physics {
              }
            }
          }
          DEF FL_upperLeg Shape {
            appearance PBRAppearance {
              baseColor 1 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.1 0.2 0.1
            }
          }
        ]
        boundingObject USE FL_upperLeg
        physics DEF upperLeg_physics Physics {
        }
      }
    }
    DEF FrontRight_upper HingeJoint {
      jointParameters HingeJointParameters {
        position -0.9
        axis 0 0 1
        anchor -0.16 0.38 -0.3
      }
      device [
        RotationalMotor {
          name "FrontRight_Shoulder"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.21645011754021948 0.32236736452500797 -0.29999977412881107
        rotation -4.061586448790542e-06 -1.962022021314982e-06 -0.999999999989827 0.9000011182754737
        children [
          DEF FrontRight_lower HingeJoint {
            jointParameters HingeJointParameters {
              position 0.8
              axis 0 0 1
              anchor 0 -0.05 -0.3
            }
            device [
              RotationalMotor {
                name "FrontRight_Elbow"
                maxTorque 100
              }
            ]
            endPoint Solid {
              translation 0.07890916999894752 -0.12663773802818823 -0.10000000000000003
              rotation 0 0 1 0.7999999999999999
              children [
                DEF FL_lowerLeg Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.25 0.1
                  }
                }
              ]
              name "solid(3)"
              boundingObject DEF FL_lowerLeg Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.1 0.25 0.1
                }
              }
              physics DEF lowerLeg_physics Physics {
              }
            }
          }
          USE FL_upperLeg
        ]
        name "solid(2)"
        boundingObject USE FL_upperLeg
        physics USE upperLeg_physics
      }
    }
    DEF BackLeft_upper HingeJoint {
      jointParameters HingeJointParameters {
        position -0.27
        axis 0 0 1
        anchor 0.16 0.38 0.3
      }
      device [
        RotationalMotor {
          name "BackLeft_Shoulder"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.12902403960955136 0.3054949023348304 0.2999997079359282
        rotation 1.4444296985278293e-05 1.9624358127978846e-06 -0.9999999998937555 0.2699964826647997
        children [
          DEF BackLeft_lower HingeJoint {
            jointParameters HingeJointParameters {
              position 0.9
              axis 0 0 1
              anchor 0 -0.05 -0.3
            }
            device [
              RotationalMotor {
                name "BackLeft_Elbow"
                maxTorque 100
              }
            ]
            endPoint Solid {
              translation 0.0861659600590232 -0.1183770965097731 0.09999999999999987
              rotation 0 0 1 0.9000000000000001
              children [
                DEF BL_lowerLeg Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.25 0.1
                  }
                }
              ]
              name "solid(5)"
              boundingObject USE BL_lowerLeg
              physics USE lowerLeg_physics
            }
          }
          DEF BL_upperLeg Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.1 0.2 0.1
            }
          }
        ]
        name "solid(4)"
        boundingObject USE BL_upperLeg
        physics USE upperLeg_physics
      }
    }
    DEF BackRight_upper HingeJoint {
      jointParameters HingeJointParameters {
        position -0.7
        axis 0 0 1
        anchor 0.16 0.38 -0.3
      }
      device [
        RotationalMotor {
          name "BackRight_Shoulder"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.10082819498161325 0.3251855777003108 -0.29999983560149984
        rotation -4.877947654234369e-06 -2.9906850062477928e-06 -0.9999999999836306 0.6999999999721774
        children [
          DEF BackRight_lower HingeJoint {
            jointParameters HingeJointParameters {
              position 1.1
              axis 0 0 1
              anchor 0 -0.05 -0.3
            }
            device [
              RotationalMotor {
                name "BackRight_Elbow"
                maxTorque 100
              }
            ]
            endPoint Solid {
              translation 0.09803280960675786 -0.09989557335681348 -0.1
              rotation 0 0 1 1.1000000000000003
              children [
                DEF BL_lowerLeg Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.1 0.25 0.1
                  }
                }
              ]
              name "solid(7)"
              boundingObject DEF BL_lowerLeg Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.1 0.25 0.1
                }
              }
              physics DEF lowerLeg_physics Physics {
              }
            }
          }
          USE BL_upperLeg
        ]
        name "solid(6)"
        boundingObject USE BL_upperLeg
        physics USE upperLeg_physics
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
    density -1
    mass 3
  }
  controller "2_joint_controller"
}
